### PR TITLE
4.2.0 JS fixes

### PR DESCRIPTION
**Pull request by qstudio**
Some of the changes in this pull request are from the version on github which was already ahead of the release on WP, the main changes we have introduced are in the JS handling of WP tinymce objects.

### CODE REVIEW

**Comment by rahularyan** [comment]
Thank you for the PR. I will check this and merge.

File: addons/profile/profile.php [hunk 1]
```diff
@@ -282,9 +282,12 @@ public function sub_page_template() {
 	 * Display user questions page.
 	 */
 	public function question_page() {
-		$user_id                        = ap_current_user_id();
+		$user_id = ap_current_user_id();
+		$paged   = ( get_query_var( 'ap_paged' ) ) ? get_query_var( 'ap_paged' ) : 1;
```
Comment (rahularyan) [inline addons/profile/profile.php:286]: Hello @qstudio 
This needs to be sanitized using `max( 1, (int) get_query_var( 'ap_paged' ) )` to make sure number is integer and not less than 1. Thanks

File: addons/syntaxhighlighter/script.js [hunk 2]
```diff
@@ -67,34 +67,46 @@ SyntaxHighlighter.all();
...
-
-      var attr = 'language="'+lang+'"';
-      var cont = '[apcode '+attr+']<'+tag+' data-mce-contenteditable="false">'+code+'</'+tag+'>[/apcode]';
-      tinymce.activeEditor.insertContent(cont);
-      tinymce.activeEditor.focus();
-      tinymce.activeEditor.selection.collapse(0);
-      AnsPress.hideModal('code');
-
-      return false;
+		var code = $(this).find('textarea').val();
```
Comment (rahularyan) [inline addons/syntaxhighlighter/script.js:70]: @qstudio Can you please retain existing indentation. Thanks

File: lib/form/class-editor.php [hunk 3]
```diff
@@ -142,6 +142,7 @@ public function field_markup() {
 		 * @since 4.1.8
 		 */
 		$value = apply_filters( 'ap_editor_pre_value', $this->value() );
+		// error_log( 'value: '.$value.' id: '.$this->id() );
```
Comment (rahularyan) [inline lib/form/class-editor.php:145]: Please remove this line.

File: includes/shortcode-basepage.php [hunk 4]
```diff
@@ -92,7 +92,7 @@ public function anspress_sc( $atts, $content = '' ) {
 		echo '</div>';
 		// Linkback to author.
 		if ( ! ap_opt( 'author_credits' ) ) {
-			echo '<div class="ap-cradit">' . esc_attr__( 'Question and answer is powered by', 'anspress-question-answer' ) . ' <a href="https://anspress.io" target="_blank">AnsPress.io</a></div>';
+			echo '<div class="ap-cradit">' . esc_attr__( 'Question and answer is powered by', 'anspress-question-answer' ) . ' <a href="https://anspress.io" traget="_blank">AnsPress.io</a></div>';
```
Comment (rahularyan) [inline includes/shortcode-basepage.php:95]: This seems to be a typo.

File: includes/hooks.php [hunk 5]
```diff
@@ -454,14 +454,12 @@ public static function publish_comment( $comment ) {
...
 		ap_activity_add( array(
-			'q_id'    => 'answer' === $post->post_type ? $post->post_parent: $post->ID,
-			'action'  => 'new_c',
-			'a_id'    => 'answer' === $post->post_type ? $post->ID: 0,
-			'c_id'    => $comment->comment_ID,
-			'user_id' => $comment->user_id,
+			'q_id'   => 'answer' === $post->post_type ? $post->post_parent: $post->ID,
+			'action' => 'new_c',
+			'a_id'   => 'answer' === $post->post_type ? $post->ID: 0,
+			'c_id'   => $comment->comment_ID,
```
Comment (rahularyan) [inline includes/hooks.php:462]: Please share more detail why you think `user_id` argument is not needed. Thanks

File: includes/functions.php [hunk 6]
```diff
@@ -254,7 +254,7 @@ function ap_human_time( $time, $unix = true, $show_full_date = 604800, $format =
 		if ( $show_full_date + $time > current_time( 'timestamp', true ) ) {
 			return sprintf(
 				/* translators: %s: human-readable time difference */
-				__( '%s ago', 'anspress-question-answer' ),
+				_x( '%s ago', 'anspress-question-answer' ),
```
Comment (rahularyan) [inline includes/functions.php:257]: `_x` second argument expect context.

File: assets/js/min/main.min.js [hunk 7]
```diff
@@ -1,3 +1,3 @@
...
...bone.View.extend({id:function(){return"noti-"+this.model.id},template:'<div class="noti-item clearfix {{seen==1 ? \'seen\' : \'unseen\'}}"><# if(ref_type === \'reputation\') { #>  <div class="ap-noti-rep<# if(points < 1) { #> negative<# } #>">{{points}}</div><# } else if(hide_actor) { #><div class="ap-noti-icon {{icon}}"></div><# } else { #><div class="ap-noti-avatar">{{{avatar}}}</div><# } #><a class="ap-noti-inner" href="{{permalink}}"><# if(ref_type !== \'reputation\'){ #><strong class="ap-not-actor">{{actor}}</strong><# } #> {{verb_label}} <strong class="ap-not-ref">{{ref_title}}</strong><time class="ap-noti-date">{{date}}</time></a></div>',initialize:function(a){this.model=a.model},render:function(){var a=_.template(this.template);return this.$el.html(a(this.model.toJSON())),this}}),AnsPress.views.Notifications=Backbone.View.extend({template:'<button class="ap-droptogg apicon-x"></button><div class="ap-noti-head">{{aplang.notifications}}<# if(total > 0) { #><i class="ap-noti-unseen">{{total}}</i><a href="#" class="ap-btn ap-btn-markall-read ap-btn-small" apajaxbtn apquery="{{JSON.stringify(mark_args)}}">{{aplang.mark_all_seen}}</a><# } #></div><div class="scroll-wrap"></div>',
```
Comment (rahularyan) [inline assets/js/min/main.min.js:1]: Please exclude this file from commit.

File: admin/anspress-admin.php [hunk 8]
```diff
@@ -1306,7 +1302,7 @@ public static function options_postscomments() {
 					'value'   => $opt['answers_sort'],
 				),
 				'minimum_ans_length'            => array(
-					'label'   => __( 'Minimum answer content', 'anspress-question-answer' ),
+					'label'   => __( 'Minimum question content', 'anspress-question-answer' ),
```
Comment (rahularyan) [inline admin/anspress-admin.php:1305]: This seems to be a typo

File: admin/anspress-admin.php [hunk 9]
```diff
@@ -1228,14 +1228,10 @@ public static function options_postscomments() {
...
-						'newest' => __( 'Newest', 'anspress-question-answer' ),
-						'voted'  => __( 'Voted', 'anspress-question-answer' ),
-						'answers' => __( 'Answers', 'anspress-question-answer' ),
-						'views' => __( 'Views', 'anspress-question-answer' ),
 						'voted'  => __( 'Voted', 'anspress-question-answer' ),
-						'unanswered' => __( 'Unanswered', 'anspress-question-answer' ),
-						'unsolved' => __( 'Unsolved', 'anspress-question-answer' ),
+						'active' => __( 'Active', 'anspress-question-answer' ),
+						'newest' => __( 'Newest', 'anspress-question-answer' ),
+						'oldest' => __( 'Oldest', 'anspress-question-answer' ),
```
Comment (rahularyan) [inline admin/anspress-admin.php:1234]: Please revert this.

**Review by rahularyan — CHANGES_REQUESTED** [review]
Hello Ray,
I have requested some changes, Please check them when you can.
Thanks.

File: addons/syntaxhighlighter/script.js (hunk 2 above)
Comment (qstudio) [reply addons/syntaxhighlighter/script.js:70]: This is not our change, I'll note all the changes in the PR which we made.

File: lib/form/class-editor.php (hunk 3 above)
Comment (qstudio) [reply lib/form/class-editor.php:145]: It is commented out, so will do no harm - do you have an internal logging method to write to the log file we can use for future testing?

File: includes/shortcode-basepage.php (hunk 4 above)
Comment (qstudio) [reply includes/shortcode-basepage.php:95]: Yep - and it will break UI, so needs fixing.

File: includes/hooks.php (hunk 5 above)
Comment (qstudio) [reply includes/hooks.php:462]: This is also not our update..

File: admin/anspress-admin.php (hunk 8 above)
Comment (qstudio) [reply admin/anspress-admin.php:1305]: Not ours..

File: assets/js/min/main.min.js (hunk 7 above)
Comment (qstudio) [reply assets/js/min/main.min.js:1]: I will have more time next week to make a cleaner commit, I will exclude this file.

File: addons/profile/profile.php (hunk 1 above)
Comment (rahularyan) [reply addons/profile/profile.php:286]: Sorry about that.

File: addons/syntaxhighlighter/script.js (hunk 2 above)
Comment (rahularyan) [reply addons/syntaxhighlighter/script.js:70]: Please ignore then

File: lib/form/class-editor.php (hunk 3 above)
Comment (rahularyan) [reply lib/form/class-editor.php:145]: Still this has to be removed. :) Thanks you

File: assets/js/min/main.min.js (hunk 7 above)
Comment (rahularyan) [reply assets/js/min/main.min.js:1]: Thanks a lot

**Comment by qstudio** [comment]
@rahularyan -- sorry for the delay, I've been busy on some other projects - what is the status of these fixes and AnsPress in general - will you be needing these fixes for a future release?

**Comment by pzg-gl** [comment]
> rahularyan -- sorry for the delay, I've been busy on some other projects - what is the status of these fixes and AnsPress in general - will you be needing these fixes for a future release?

Hello @qstudio, 

We are actively working on AnsPress development again. However, would it be possible for you to send a new pull request? The old one got mixed up. It should clear some things up!

**Pull request closed without merging**